src/main/java/Parser.java

```diff
@@ -0,0 +1,94 @@
+public class Parser {
+    private String input;
+    private boolean isAddingTask;
+    private boolean isMarkingTask;
+    private boolean isListingTasks;
+    private boolean isExiting;
+
+    Parser() {
+        this.input = "";
+        this.isAddingTask = false;
+        this.isMarkingTask = false;
+        this.isListingTasks = false;
+        this.isExiting = false;
+    }
+
+    private void reset() {
+        input = "";
+        isAddingTask = false;
+        isMarkingTask = false;
+        isListingTasks = false;
+        isExiting = false;
+    }
+
+    public void parseString(String input) {
+        reset();
+        this.input = input;
+        String command = getCommand();
+        setParam(command);
+    }
+
+    private void setParam(String command) {
+        if (command.equals("bye")) {
+            isExiting = true;
+        } else if (command.equals("list")) {
+            isListingTasks = true;
+        } else if (command.equals("mark") || command.equals("unmark")) {
+            isMarkingTask = true;
+        } else if (command.equals("todo") || command.equals("event")
+                || command.equals("deadline")) {
+            isAddingTask = true;
+        }
```

Comment:
Good job on encasing all lines with curly braces even if the statements are single lines